common/workunit/workunit.cpp

```diff
@@ -13526,12 +13526,30 @@ IPropertyTree * resolveDefinitionInArchive(IPropertyTree * archive, const char *
 
 extern WORKUNIT_API void associateLocalFile(IWUQuery * query, WUFileType type, const char * name, const char * description, unsigned crc, unsigned minActivity, unsigned maxActivity)
 {
-    StringBuffer hostname;
-    queryHostIP().getIpText(hostname);
-
-    StringBuffer fullPathname;
-    makeAbsolutePath(name, fullPathname);
-    query->addAssociatedFile(type, fullPathname, hostname, description, crc, minActivity, maxActivity);
+    StringBuffer fullPathName;
+    makeAbsolutePath(name, fullPathName);
+    if (isCloud())
+    {
+        const char *dllserver_root = getenv("HPCC_DLLSERVER_PATH");
+        assertex(dllserver_root != nullptr);
+        StringBuffer destPathName(dllserver_root);
+        addNonEmptyPathSepChar(destPathName);
+        splitFilename(fullPathName.str(), nullptr, nullptr, &destPathName, &destPathName);
+        OwnedIFile source = createIFile(fullPathName);
+        OwnedIFile target = createIFile(destPathName);
+        if (!target->exists())
```

Comment:
@richardkchapman I'll merge as-is, but if this was an s3 bucket then this might add a noticable latency.

Reply:
Ok, but some files do get added twice (once as a "helper" and once as a "dll"), so will want to refactor elsewhere to avoid that